android/src/main/java/me/furtado/smsretriever/SmsHelper.java

```diff
@@ -41,46 +53,90 @@ void startRetriever(final Promise promise) {
             return;
         }
 
-        final SmsRetrieverClient client = SmsRetriever.getClient(mContext);
-        final Task<Void> task = client.startSmsRetriever();
-        task.addOnSuccessListener(mOnSuccessListener);
-        task.addOnFailureListener(mOnFailureListener);
+        registerReceiver();
+        startSmsRetrieverClient();
+    }
+
+    public void onSmsReceived(final String message) {
+        if (message == null || message.isEmpty()) {
+            Log.w(TAG, "Received empty sms message");
+        }
+        else {
+            Log.d(TAG, "Received sms message");
+            mNumMessages -= 1;
```

Comment:
Make `mNumMessages` AtomicInteger to avoid rare race condition since `-=` is not atomic.

Reply:
Good thinking but I think it's unnecessary in this case because the API we're relying on is guaranteed to only provide one SMS per client instance, so this function should (theoretically) never be called twice in parallel